code/game/atoms_movable.dm

```diff
@@ -24,6 +24,7 @@
 	glide_size = 8
 	appearance_flags = TILE_BOUND
 	var/datum/forced_movement/force_moving = null	//handled soley by forced_movement.dm
+	var/atom/hit_obj = null

```

Comment:
really? you're storing the object it hit on the atom instead of say passing it through to the appropriate methods as a variable? You're gonna need to justify this